zebra-chain/src/block.rs

```diff
@@ -169,6 +169,12 @@ pub struct Block {
     pub transactions: Vec<Transaction>,
 }
 
+impl<'a> From<&'a Block> for BlockHeaderHash {
+    fn from(block: &'a Block) -> BlockHeaderHash {
+        (&block.header).into()
```

Comment:
Why are the extra parens needed vs just `&block.header.into()`?

Reply:
The problem is that `&block.header.into()` will make a `&T`, where `impl Into<T> for BlockHeader`, but we want `T` where `impl Into<T> for &BlockHeader`, so the & has to be inside the parentheses.